identity/src/androidTest/java/com/android/identity/IdentityCredentialTest.java

```diff
@@ -0,0 +1,172 @@
+  }
+
+  /**
+   * Tests storing static authentication data of varying size for varying numbers of auth keys.
+   */
+  @MediumTest
+  @Test
+  public void storeStaticAuthenticationData()
+          throws IdentityCredentialException {
+    checkStaticAuthData(1, 100); // 1 key x 100 bytes of static authentication data = 100 Bytes
+    checkStaticAuthData(3, 100);
+    checkStaticAuthData(100, 100); // 100 keys x 100 Bytes = 10 KBytes
+    checkStaticAuthData(1, 100000); // 1 key x 100 KBytes
+  }
+
+  @LargeTest
+  @Test
+  public void storeStaticAuthenticationDataLarge()
+          throws IdentityCredentialException {
+    Assume.assumeTrue("Test fails on API <= 28", VERSION.SDK_INT >= 29 /* Android 10+ */);
+    // Per b/234563696, I've confirmed these failing on:
+    // Pixel 2 API 24
+    // Pixel 2 API 26
+    // Pixel 3 API 28
+    checkStaticAuthData(10, 10000); // fails
+    checkStaticAuthData(100, 1000); // flaky
+    checkStaticAuthData(100, 10000); // fails
+  }
+
+  private void checkStaticAuthData(int numAuthKeys, int staticAuthDataSizeBytes)
+          throws IdentityCredentialException {
+    final int usesPerKey = 3;
+    ProvisioningTest.createCredential(store, CREDENTIAL_NAME);
+    try {
+      IdentityCredential credential = Preconditions.checkNotNull(getCredentialIfPresent());
+
+      storeFakeStaticAuthData(credential, usesPerKey, numAuthKeys, staticAuthDataSizeBytes);
+
+      // Check that this doesn't throw (it did throw in internal b/234563696).
+      try {
+        getCredentialIfPresent();
+      } catch (RuntimeException e) {
+        String msg = String.format(Locale.US,
+                "Failed to load credential for %d authKeys with %d bytes static auth data each",
+                numAuthKeys, staticAuthDataSizeBytes);
+        throw new IdentityCredentialException(msg, e);
+      }
+    } finally {
+      deleteCredentialFromStoreIfPresent();
+    }
+  }
+
+  @Before
+  public void setUp() {
+    deleteCredentialFromStoreIfPresent(); // clean up in case a previous test didn't
+  }
+
+  @After
+  public void tearDown() {
+    deleteCredentialFromStoreIfPresent();
+    store = null;
+  }
+
+  private static void storeFakeStaticAuthData(IdentityCredential credential,
+          int maxUsesPerKey, int numKeys, int staticAuthDataSizeBytes) {
+    Random random = new Random(31337 + numKeys);
+    credential.setAvailableAuthenticationKeys(numKeys, maxUsesPerKey);
+    Collection<X509Certificate> authKeys = credential.getAuthKeysNeedingCertification();
+    assertEquals(numKeys, authKeys.size());
+
+    for (X509Certificate authKey : authKeys) {
+      byte[] staticAuthData = new byte[staticAuthDataSizeBytes];
+      random.nextBytes(staticAuthData);
+      Calendar expirationDate = Calendar.getInstance();
+      expirationDate.add(Calendar.YEAR, 1);
+      try {
+        credential.storeStaticAuthenticationData(authKey, expirationDate, staticAuthData);
+      } catch (UnknownAuthenticationKeyException e) {
```

Comment:
Another case where I think it's simpler to just let junit handle this exception for us.

Reply:
Done